log_drains.go

```diff
@@ -23,3 +39,44 @@ func (c *Client) LogDrainsList(app string) ([]LogDrain, error) {
 	}
 	return logDrainsRes, nil
 }
+
+type LogDrainAddParams struct {
+	Type        string `json:"type"`
+	URL         string `json:"url"`
+	Port        string `json:"port"`
+	Host        string `json:"host"`
+	Token       string `json:"token"`
+	DrainRegion string `json:"drain_region"`
+}
+
+func (c *Client) LogDrainAdd(app string, params LogDrainAddParams) (*LogDrainRes, error) {
+	var logDrainRes LogDrainRes
+	payload := logDrainReq{
+		Drain: LogDrain{
+			Type:        params.Type,
+			URL:         params.URL,
+			Host:        params.Host,
+			Port:        params.Port,
+			Token:       params.Token,
+			DrainRegion: params.DrainRegion,
+		},
+	}
+
+	req := &httpclient.APIRequest{
+		Method:   "POST",
+		Endpoint: "/apps/" + app + "/log_drains",
+		Expected: httpclient.Statuses{201, 422},
+		Params:   payload,
+	}
+
+	err := c.ScalingoAPI().DoRequest(req, &logDrainRes)
```

Comment:
I think you should use `SubresourceAdd` instead of `DoRequest` here, don't you think? Because here, you wrote `Expected: httpclient.Statuses{201, 422},` which means that you don't return an error if the API returns 422. But don't you want to return an error in such cases?

Reply:
Yes, I want to return an error with the error message from the API.
That's why I expect for a 422, to be able to read the response body (especially error field from json)